mesosutil/mesosprotoutil.go

```diff
@@ -36,28 +36,49 @@ func FilterResources(resources []*mesos.Resource, filter func(*mesos.Resource) b
 	return result
 }
 
-func NewScalarResource(name string, val float64) *mesos.Resource {
-	return &mesos.Resource{
+// A ResourceOpt is a functional option type for mesos.Resources.
+type ResourceOpt func(*mesos.Resource)
+
+// Role returns a ResourceOpt that sets the Role of a
+// *mesos.Resource to the given value.
+func Role(s *string) ResourceOpt {
+	return func(r *mesos.Resource) { r.Role = s }
+}
+
+func NewScalarResource(name string, val float64, resOpt ...ResourceOpt) *mesos.Resource {
```

Comment:
At this stage, it's  OK to break the API for the sake of consistency. Users of this library need to vendor it to get reproducible builds. Hence, I'd like to have a single `func NewResource(opts ...ResourceOpt) *mesos.Resource` and create the necessary `ResourceOpts` to set every resource field.